tests/std/tests/Dev10_555491_complex_linker_errors/test.cpp

```diff
@@ -90,4 +95,36 @@ int main() {
     assert(nearly_equal_partwise(asinh(-1e+307 + 2e+307i), -708.3914896859491 + 1.1071487177940904i));
     assert(nearly_equal_partwise(asinh(-2e+37f + 4e+37if), -87.386663f + 1.1071488if));
     assert(nearly_equal_partwise(asinh(-1e+307L + 2e+307il), -708.3914896859491L + 1.1071487177940904il));
+
+    // Also test GH-785 "<complex>: std::arg does not work for negative real values"
+
+    STATIC_ASSERT(is_same_v<decltype(arg(1729)), double>);

```

Comment:
Also test float and long double which shouldn't get upgraded?

Reply:
Good idea, done!